src/Analysis/pipeline/parameter_types.py

```diff
@@ -0,0 +1,323 @@
+import abc
+
+class BaseData(metaclass=abc.ABCMeta):
+    """
+    Abbstract class od port types, that define user for
+    validation and translation to string
+    """
+    def __init__(self):
+        self.name = ""
+        """Display name of port"""
+        self.description = ""
+        """Display description of port"""
+    
+    @abc.abstractmethod 
+    def to_string(self, value):
+        """Presentation of type in json yaml"""
+        pass
+
+    def contain(self, data):
+        """
+        return if structure contain set data
+        """
+        return True
+        
+    @abc.abstractmethod 
```

Comment:
Typ přejmenován, popis doplňen, jen si nejsem jistý is_set => is_data_tree, máto sloužit k rozeznání, zda ve struktuře jsou již reálný data. (V budoucnu, pokud se to povede, k tomu jestli tam reálný data mohou probublat přes předky) Možná by se to mělo jmenovat úplně jinak?